### PR TITLE
Merge mp changes

### PR DESCRIPTION
**Pull request by JonathanSchmidt1**
## Summary

Include a summary of major changes in bullet points:

* Feature 1
* Fix 1

## Additional dependencies introduced (if any)

* List all new dependencies needed and justify why. While adding dependencies that bring
  significantly useful functionality is perfectly fine, adding ones that add trivial
  functionality, e.g., to use one single easily implementable function, is frowned upon.
  Justify why that dependency is needed. Especially frowned upon are circular dependencies.

## TODO (if any)

If this is a work-in-progress, write something about what else needs to be done.

* Feature 1 supports A, but not B.

## Checklist

Work-in-progress pull requests are encouraged, but please put [WIP] in the pull request
title.

Before a pull request can be merged, the following items must be checked:

* [ ] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
  The easiest way to handle this is to run the following in the **correct sequence** on
  your local machine. Start with running [black](https://black.readthedocs.io/en/stable/index.html) on your new code. This will
  automatically reformat your code to PEP8 conventions and removes most issues. Then run
  [ruff](https://ruff.rs).
* [ ] Docstrings have been added in the [Numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
  Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [ ] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org) to
  type check your code.
* [ ] Tests have been added for any new functionality or bug fixes.
* [ ] All linting and tests pass.

Note that the CI system will run all the above checks. But it will be much more
efficient if you already fix most errors prior to submitting the PR. It is highly
recommended that you use the pre-commit hook provided in the repository. Simply run
`pre-commit install` and a check will be run prior to allowing commits.

### CODE REVIEW

File: src/atomate2/common/jobs/phonons.py [hunk 1]
```diff
@@ -159,19 +161,19 @@ def generate_phonon_displacements(
...
     """
-    cell = get_phonopy_structure(structure)
+    cell = magnetic_get_phonopy_structure(structure)
     if code == "vasp":
         factor = VaspToTHz
     # a bit of code repetition here as I currently
     # do not see how to pass the phonopy object?
-    if use_symmetrized_structure == "primitive" and kpath_scheme != "seekpath":
-        primitive_matrix: list[list[float]] | str = [
+#    if use_symmetrized_structure == "primitive" and kpath_scheme != "seekpath":
```
Comment (JaGeo) [inline src/atomate2/common/jobs/phonons.py:169]: I don't think this is correct for seekpath. Could you elaborate why you removed the other options?

File: src/atomate2/forcefields/jobs.py [hunk 2]
```diff
@@ -26,8 +26,10 @@
     "CHGNetRelaxMaker",
     "M3GNetStaticMaker",
     "M3GNetRelaxMaker",
-    "GAPRelaxMaker",
```
Comment (JaGeo) [inline src/atomate2/forcefields/jobs.py:29]: Just make sure this isn't removed

File: src/atomate2/common/schemas/phonons.py [hunk 3]
```diff
@@ -372,31 +375,32 @@ def from_forces_born(
...
         heat_capacities = [
-            dos.cv(structure=get_pmg_structure(phonon.primitive), t=temperature)
+            dos.cv(structure=magnetic_get_pmg_structure(phonon.primitive), t=temperature)
             for temperature in temperature_range
         ]
 
         # will compute thermal displacement matrices
         # for the primitive cell (phonon.primitive!)
         # only this is available in phonopy
+        kwargs['create_thermal_displacements'] = False
```
Comment (JaGeo) [inline src/atomate2/common/schemas/phonons.py:403]: The default is set a line below.

**Comment by JaGeo** [comment]
@JonathanSchmidt1 Thanks for this. I assume it is still work in progress. Since I saw it, I added a few comments/suggestions/ideas.

It might be worth adding some of the functions to pymatgen (magnetic_get_pmg_structure etc.).

File: src/atomate2/common/utils.py [hunk 4]
```diff
@@ -11,6 +11,65 @@
 if TYPE_CHECKING:
     from pathlib import Path
 
+from pymatgen.io.phonopy import get_phonopy_structure
```
Comment (JaGeo) [inline src/atomate2/common/utils.py:14]: I think this should be fixed in pymatgen directly. @janosh what do you think?

Comment (janosh) [reply src/atomate2/common/utils.py:14]: Agreed. PR would be great 👍

File: src/atomate2/common/schemas/math.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+"""Schemas for matrices and vectors."""
+
+from typing import Tuple
+
+__all__ = ["Vector3D", "Vector6D", "Matrix3D", "MatrixVoigt"]
+
+
+Vector3D = Tuple[float, float, float]
```
Comment (utf) [inline src/atomate2/common/schemas/math.py:8]: These are already in emmet: https://github.com/materialsproject/emmet/blob/main/emmet-core/emmet/core/math.py

File: src/atomate2/common/schemas/structure.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+from pymatgen.core.periodic_table import Element
+
+from atomate2.common.schemas.symmetry import SymmetryData
+
+__all__ = ["StructureMetadata"]
+
+T = TypeVar("T", bound="StructureMetadata")
+
+
+class StructureMetadata(BaseModel):
```
Comment (utf) [inline src/atomate2/common/schemas/structure.py:16]: These schemas are also already in emmet.

File: src/atomate2/common/schemas/symmetry.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+from pydantic import BaseModel, Field
+from pymatgen.core import Structure
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer, spglib
+
+from atomate2 import SETTINGS
+
+__all__ = ["CrystalSystem", "SymmetryData"]
+
+
+class CrystalSystem(ValueEnum):
```
Comment (utf) [inline src/atomate2/common/schemas/symmetry.py:15]: These schemas are also already in emmet.

File: src/atomate2/vasp/jobs/phonons.py [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+    "get_supercell_size",
+    "generate_phonon_displacements",
+    "run_phonon_displacements",
+    "generate_frequencies_eigenvectors",
+    "PhononDisplacementMaker",
+]
+
+
+@job
+def get_total_energy_per_cell(
```
Comment (utf) [inline src/atomate2/vasp/jobs/phonons.py:39]: Most (if not all) of these functions already exist in: https://github.com/materialsproject/atomate2/blob/main/src/atomate2/common/jobs/phonons.py

File: src/atomate2/vasp/schemas/calculation.py [hunk 9]
```diff
@@ -0,0 +1,924 @@
...
+    "IonicStep",
+    "ElectronicStep",
+    "ElectronPhononDisplacedStructures",
+]
+
+
+_BADER_EXE_EXISTS = bool(which("bader") or which("bader.exe"))
+
+
+class Status(ValueEnum):
```
Comment (utf) [inline src/atomate2/vasp/schemas/calculation.py:61]: These are also in emmet.

**Review by utf — COMMENTED** [review]
Thanks for this @JonathanSchmidt1. For now, my main comment is that you have re-added a lot of Schema's back to atomate2 but these should instead be imported from emmet.

**Comment by JonathanSchmidt1** [comment]
Really sorry everyone this was pushed to the wrong base repository.

**Pull request closed without merging**